### PR TITLE
Ensure dirty-bit is updated when calling save on a syncable model with `update_fields`

### PR DESCRIPTION
**Pull request by bjester**
## Summary
- Adds regression test that verified the issue scenario
- Addresses the issue scenario by adding the dirty-bit field to `update_fields` when present
- Bumps version and updates changelog

## TODO

- [X] Have tests been written for the new code?
- [ ] Has documentation been written/updated?
- [ ] New dependencies (if any) added to requirements file

## Reviewer guidance
- Does the regression test make sense?

## Issues addressed

closes https://github.com/learningequality/morango/issues/325

## AI usage
The fix was pretty clear, but I let AI take the wheel. It wrote the regression test, then we verified the issue, then proceeded with the fix.

### CODE REVIEW

File: morango/models/core.py [hunk 1]
```diff
@@ -849,6 +849,13 @@ def save(self, update_dirty_bit_to=True, *args, **kwargs):
             self._morango_dirty_bit = True
         elif not update_dirty_bit_to:
             self._morango_dirty_bit = False
+
+        # ensure the dirty bit field is in the fields to update if present, to keep it in sync
+        if update_dirty_bit_to is not None and kwargs.get("update_fields") is not None:
```
Comment (rtibblesbot) [inline morango/models/core.py:854]: **praise:** The `update_dirty_bit_to is not None` guard correctly handles the opt-out case — callers who pass `None` to suppress all dirty-bit handling also suppress the `update_fields` injection, which is exactly right.

File: tests/testapp/tests/integration/test_syncing_models.py [hunk 2]
```diff
@@ -67,6 +67,15 @@ def test_syncable_save(self):
...
         self.assertTrue(MyUser.objects.first()._morango_dirty_bit)
 
+    def test_syncable_save_with_update_fields_persists_dirty_bit(self):
+        user = MyUser.objects.first()
+        user.save(update_dirty_bit_to=False)
+        self.assertFalse(MyUser.objects.first()._morango_dirty_bit)
+
+        user.username = "updated-name"
+        user.save(update_fields=["username"])
+        self.assertTrue(MyUser.objects.first()._morango_dirty_bit)
```
Comment (rtibblesbot) [inline tests/testapp/tests/integration/test_syncing_models.py:77]: **suggestion:** This test covers the common case well. Consider adding a second assertion for `save(update_dirty_bit_to=None, update_fields=["username"])` to verify that the dirty bit is NOT updated in the DB when callers explicitly opt out — the fix handles it correctly, but a test would prevent regression if the guard condition is ever changed.

**Review by rtibblesbot — COMMENTED** [review]
Clean, minimal fix for a nasty P0 edge case.

CI passing. No UI files changed — visual inspection skipped.

- **suggestion:** `release-v0.9.x` has the same unfixed `save` method (confirmed by reading the branch). Since this is P0, a follow-up PR or backport to the default branch is worth tracking.
- **suggestion:** (inline) consider a test for the `update_dirty_bit_to=None` + `update_fields` opt-out path
- **praise:** (inline) the `update_dirty_bit_to is not None` guard correctly scopes the fix

---
_@rtibblesbot's comments are generated by an LLM, and should be evaluated accordingly_

<details><summary>How was this generated?</summary>

Reviewed the pull request diff checking for:
- **Correctness:** bugs, edge cases, undocumented behavior, resource leaks, hardcoded values
- **Design:** unnecessary complexity, naming, readability, comment accuracy, redundant state
- **Architecture:** duplicated concerns, minimal interfaces, composition over inheritance
- **Testing:** behavior-based assertions, mocks only at hard boundaries, accurate coverage
- **Completeness:** missing dependencies, unupdated usages, i18n, accessibility, security
- **Principles:** DRY (same reason to change), SRP, Rule of Three (no premature abstraction)
- Checked CI status and linked issue acceptance criteria
- For UI changes: inspected screenshots for layout, visual completeness, and consistency

</details>

**Review by ozer550 — APPROVED** [review]
This LGTM!

File: tests/testapp/tests/integration/test_syncing_models.py [hunk 3]
```diff
@@ -67,6 +67,24 @@ def test_syncable_save(self):
...
+
+    def test_syncable_save_with_update_fields_when_to_ignore_dirty_bit(self):
```
Comment (rtibblesbot) [inline tests/testapp/tests/integration/test_syncing_models.py:79]: **praise:** Naming clearly isolates the opt-out scenario from the main regression test, making the intent self-documenting — good application of the single-responsibility principle to tests.

**Review by rtibblesbot — COMMENTED** [review]
### Prior findings

**Resolved:**
- Add test for `update_dirty_bit_to=None` + `update_fields` case (suggestion) — `test_syncable_save_with_update_fields_when_to_ignore_dirty_bit` added

**Unaddressed (re-raised below):**
- `release-v0.9.x` has the same unfixed `save()` method (suggestion) — no reply, fix not found on that branch

1/2 prior findings resolved. 1 re-raised below.

---

Python tests queued for `8106e8d`; migrations passed. Prior commit passed all CI. No UI changes — Phase 3 skipped.

- **suggestion (re-raised):** `release-v0.9.x` still has the same unfixed `save()` — a backport PR or follow-up issue would ensure this P0 fix lands on the default branch.
- **praise:** Noted inline.

---
_@rtibblesbot's comments are generated by an LLM, and should be evaluated accordingly_

<details><summary>How was this generated?</summary>

Compared the current PR state against findings from a prior review:
- Retrieved prior bot reviews deterministically via the GitHub API
- Classified each prior finding as RESOLVED, UNADDRESSED, ACKNOWLEDGED, or CONTESTED
- Only raised NEW findings for newly introduced code
- Reviewed the pull request diff checking for correctness, design, architecture, testing, completeness, and adherence to DRY/SRP principles
- Checked CI status and linked issue acceptance criteria

</details>
